### ISSUE TITLE
Appears to wrap other functions than createClass()

### ISSUE THREAD

**Issue by gaearon** [gaearon/babel-plugin-react-transform#84]
As reported in https://github.com/gaearon/react-transform-catch-errors/issues/25, it seems that the plugin regressed in 2.x and thinks `React.shape` is something like `React.createClass` just because it receives an object as the argument.

**Comment by gaearon** [comment]
I think the problem is here.

Most likely that these two lines should end with `||` rather than `&&`:

https://github.com/gaearon/babel-plugin-react-transform/blob/63cafc90625c99783d4ba793d58bb59523624364/src/index.js#L77

https://github.com/gaearon/babel-plugin-react-transform/blob/63cafc90625c99783d4ba793d58bb59523624364/src/index.js#L122

Please feel free to submit a fix, add tests for it, and I’ll be happy to merge.

**Comment by nfcampos** [comment]
Why is the expected behaviour in this test wrapping the class? Not all classes with a render method are React components
https://github.com/gaearon/babel-plugin-react-transform/blob/master/test/fixtures/code-class-with-render-method/actual.js

**Comment by gaearon** [comment]
We should expect that

```js
class Foo {
  render() {}
}
```

does **not** get wrapped.

Same for

```js
var Foo = PropTypes.shape({ 
  render: PropTypes.func
})
```

It also should not be wrapped.

**Comment by nfcampos** [comment]
Exactly, that's why this test can't be right https://github.com/gaearon/babel-plugin-react-transform/blob/master/test/fixtures/code-class-with-render-method/expected.js

Also, none of these should be wrapped either, right?
https://github.com/gaearon/babel-plugin-react-transform/blob/master/test/fixtures/code-call-expression-with-render-method/actual.js

**Comment by gaearon** [comment]
>Exactly, that's why this test can't be right

It’s wrong.

>Also, none of these should be wrapped either, right?

No. However they should be wrapped if `factory` is specified as one of the values of the optional `factoryMethods` array config option.

**Comment by nfcampos** [comment]
It looks like to me that this line shouldn't even be there https://github.com/gaearon/babel-plugin-react-transform/blob/63cafc90625c99783d4ba793d58bb59523624364/src/index.js#L78
We never want to wrap a class that doesn't extend React.Component (we might if we wanted to support extending a class that itself extends React.Component but whatever), right? If it doesn't pass that superclass test there's really no reason to check whether it has a render method is there?

**Comment by gaearon** [comment]
If you change it to be `||` then I think it makes sense. "If either it's not a descendant -or- it doesn't have a render method, bail out because it is definitely not something we can wrap."

**Comment by nfcampos** [comment]
the goal:
```js
if (shouldNotWrap)
  return
else
  wrap()
```
a) current implementation:
```js
if (doesntExtendSuperClass && doesntHaveRenderMethod)
  return
else
  wrap()
// !(doesntExtendSuperClass && doesntHaveRenderMethod) === (extendsSuperClass || hasRenderMethod)
```
b) change to `||`:
```js
if (doesntExtendSuperClass || doesntHaveRenderMethod)
  return
else
  wrap()
// !(doesntExtendSuperClass || doesntHaveRenderMethod) === (extendsSuperClass && hasRenderMethod)
// so this would only wrap descendants of the super class that have a render method
// which makes all the current tests (but one) fail
// but is maybe what we want actually, seeing as the render() method is required in react classes (see https://facebook.github.io/react/docs/component-specs.html#render
```
c) remove the render method check:
```js
if (doesntExtendSuperClass)
  return
else
  wrap()
// !(doesntExtendSuperClass) === (extendsSuperClass)
// so this would wrap all classes that are descendants of the super class
// which fails only 2 of the current tests (the 2 tests that are expecting the wrong thing)
```
so which one do we want? (b) or (c)?

**Comment by gaearon** [comment]
I think we want (b). Only descendants *with* render methods are good to go.

**Comment by gaearon** [comment]
Should be fixed in 2.0.1.
